plugins/CoreVisualizations/javascripts/jqplotEvolutionGraph.js

```diff
@@ -121,6 +122,10 @@
                                 if (period) {
                                     url += '&date=' + date;
                                 }
+
+                                if (segment) {
+                                    url += '&segment=' + encodeURIComponent(decodeURIComponent(segment)); // ensure segment is encoded once
```

Comment:
Could you check that this works also when a segment contains a comma `,` and on both firefox and chrome? as we have this weird code here: https://github.com/piwik/piwik/blob/2.16.2/plugins/SegmentEditor/javascripts/Segmentation.js#L1357-L1359 i'm wondering whether this works the same on all browsers...